pvacseq/lib/generate_protein_fasta.py

```diff
@@ -0,0 +1,106 @@
+import sys
+from pathlib import Path # if you haven't already done so
+root = str(Path(__file__).resolve().parents[1])
+sys.path.append(root)
+import argparse
+import tempfile
+import os
+import yaml
+from collections import OrderedDict
+import lib
+
+def define_parser():
+    parser = argparse.ArgumentParser("pvacseq generate_protein_fasta")
+
+    parser.add_argument(
+        "input_file",
+        help="A VEP-annotated single-sample VCF containing transcript, Wildtype protein sequence, and Downstream protein sequence information"
+    )
+    parser.add_argument(
+        "peptide_sequence_length", type=int,
+        help="Length of the peptide sequence to use when creating the FASTA.",
+    )
+    parser.add_argument(
+        "output_file",
+        help="The output fasta file"
+    )
+    parser.add_argument(
+        "-d", "--downstream-sequence-length",
+        default='1000',
+        help="Cap to limit the downstream sequence length for frameshifts when creating the fasta file. "
+            + "Use 'full' to include the full downstream sequence. Default: 1000"
+    )
+    return parser
+
+def convert_vcf(input_file, temp_dir):
+    print("Converting VCF to TSV")
+    tsv_file = os.path.join(temp_dir, 'tmp.tsv')
+    convert_params = [
+        input_file,
+        tsv_file,
+    ]
+    lib.convert_vcf.main(convert_params)
+    print("Completed")
+
+def generate_fasta(peptide_sequence_length, downstream_sequence_length, temp_dir):
+    print("Generating Variant Peptide FASTA and Key File")
+    tsv_file = os.path.join(temp_dir, 'tmp.tsv')
+    fasta_file = os.path.join(temp_dir, 'tmp.fasta')
+    fasta_key_file = os.path.join(temp_dir, 'tmp.fasta.key')
+    generate_fasta_params = [
+        tsv_file,
+        str(peptide_sequence_length),
+        "15",
```

Comment:
Ok, I will change it to 0 then. Just have to be aware that this command might output more fasta sequences than the ones used produced during the full pipeline run.